### DISCUSSION TITLE
Override GitHub's url when retrieving releases notes

### DISCUSSION THREAD

**Discussion by Romain0424** [renovatebot/renovate#25469] [Suggest an Idea]
### Tell us more.
Hello everyone,

We would like to be able to override GitHub's url when retrieving releases notes. In our context, flows to GitHub are forbiden and everything is proxyfied with Artifactory.

We would like to set `github_url="https://our-artifactory-instance.fr/github-releases-notes"` for example.

Many thanks !

**Comment by HonkingGoose** [comment]
[Renovate's `customChangelogUrl` docs](https://docs.renovatebot.com/configuration-options/#customchangelogurl) says:

> **Note**
> Renovate can fetch changelogs from Bitbucket, Gitea (Forgejo), GitHub and GitLab platforms only, and setting the URL to an unsupported host/platform type won't change that.

So fetching changelogs from Artifactory won't work, even if you set a `customChangelogUrl`.

---

**Edit:** I just noticed this is a feature request type question. I'll let the maintainers answer the feature request.

**Reply by gregoryboue** [reply]
It can work in our case because in our private company network, only Artifactory has access to github and CI/CD pipelines using Renovate has access to Artifactory

**Comment by rarkins** [comment]
@viceice do you think it could work if we added a new option to hostRules called something like `replaceHost` and it basically does a search/replace from matchHost to replaceHost?

**Reply by viceice** [reply]
maybe 🤷‍♂️

**Reply by viceice** [reply]
@rarkins WDYT about supporting the same url replace which we do on containerbase? this would solve a lot use cases. 🤔

**Reply by rarkins** [reply]
I have a simpler alternative using matchHost+replaceHost in hostRules, will push a WIP soon
